File: Wavefront-DT/aspnetcore-app/src/BeachShirts.Common/Tracing.cs
using System;
using Microsoft.AspNetCore.Http;
using OpenTracing;
using OpenTracing.Noop;
using OpenTracing.Propagation;
using OpenTracing.Tag;

namespace BeachShirts.Common
{
    public static class Tracing
    {
        public static ITracer Init(string service)
        {
            // TODO: Replace this with Wavefront Tracer
            return NoopTracerFactory.Create();
        }

        public static IScope StartServerSpan(
            ITracer tracer, HttpContext context, string operationName)
        {
            ISpanBuilder spanBuilder;
            try
            {
                ISpanContext parentSpanCtx = tracer.Extract(BuiltinFormats.HttpHeaders,
                    new RequestHeadersExtractAdapter(context.Request.Headers));
                if (parentSpanCtx == null)
                {
                    spanBuilder = tracer.BuildSpan(operationName);
                }
                else
                {
                    spanBuilder = tracer.BuildSpan(operationName).AsChildOf(parentSpanCtx);
                }
            }
            catch (ArgumentException)
            {
                spanBuilder = tracer.BuildSpan(operationName);
            }
            return spanBuilder.WithTag(Tags.SpanKind.Key, Tags.SpanKindServer).StartActive(true);
        }
    }
}

File: Wavefront-DT/dropwizard-app/common/src/main/java/com/wfsample/common/Tracing.java
package com.wfsample.common;

import io.opentracing.Scope;
import io.opentracing.SpanContext;
import io.opentracing.Tracer;
import io.opentracing.noop.NoopTracerFactory;
import io.opentracing.propagation.Format;
import io.opentracing.propagation.TextMapAdapter;
import io.opentracing.tag.Tags;

import javax.ws.rs.core.MultivaluedMap;
import java.util.HashMap;

public final class Tracing {
  private Tracing() {
  }

  public static Tracer init(String service) {
    // TODO: Replace this with Wavefront Tracer
    return NoopTracerFactory.create();
  }

  public static Scope startServerSpan(Tracer tracer, javax.ws.rs.core.HttpHeaders httpHeaders, String operationName) {
    // format the headers for extraction
    MultivaluedMap<String, String> rawHeaders = httpHeaders.getRequestHeaders();
    final HashMap<String, String> headers = new HashMap<>();
    for (String key : rawHeaders.keySet()) {
      headers.put(key, rawHeaders.get(key).get(0));
    }

    Tracer.SpanBuilder spanBuilder;
    try {
      SpanContext parentSpanCtx = tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMapAdapter(headers));
      if (parentSpanCtx == null) {
        spanBuilder = tracer.buildSpan(operationName);
      } else {
        spanBuilder = tracer.buildSpan(operationName).asChildOf(parentSpanCtx);
      }
    } catch (IllegalArgumentException e) {
      spanBuilder = tracer.buildSpan(operationName);
    }
    return spanBuilder.withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER).startActive(true);
  }
}
